pkg/http/utils.go

```diff
@@ -77,17 +84,88 @@ func ToHttpRequest(requestFlag string) (Request, error) {
 	}, nil
 }
 
-func interpolateDates(source string) string {
-	return todayTemplateRegex.ReplaceAllStringFunc(source, func(templateString string) string {
-		offsetDays := 0
+func legacyDateElements(source string) string {
+	offsetDays := 0
 
-		if templateString == "{tomorrow}" {
-			offsetDays = 1
-		} else if extractedOffset := todayTemplatePlusMinusRegex.FindString(templateString); len(extractedOffset) > 0 {
-			offsetDays, _ = strconv.Atoi(extractedOffset)
-		}
+	if source == "{today}" {
+		offsetDays = 0
+	} else if source == "{tomorrow}" {
+		offsetDays = 1
+	} else if extractedOffset := templatePlusMinusRegex.FindString(source); len(extractedOffset) > 0 {
+		offsetDays, _ = strconv.Atoi(extractedOffset)
+	}
+
+	// the date below is how the golang date formatter works. it's used for the formatting. it's not what is actually going to be displayed
+	return time.Now().Add(time.Duration(offsetDays) * 24 * time.Hour).Format("2006-01-02")
+}
+
+func dateElements(source string) string {
+	r := templateDatesRegex.FindStringSubmatch(source)
+
+	if r == nil {
+		return source
+	}
+	days := r[1]
+	months := r[2]
+	years := r[3]
+
+	offsetDays, _ := strconv.Atoi(days)
+	offsetMonths, _ := strconv.Atoi(months)
+	offsetYears, _ := strconv.Atoi(years)
+
+	// the date below is how the golang date formatter works. it's used for the formatting. it's not what is actually going to be displayed
+	return time.Now().AddDate(offsetYears, offsetMonths, offsetDays).Format("2006-01-02")
+}
+
+func timestampElements() string {
+	epoch := time.Now().UnixNano() / 1000000
+
+	return strconv.FormatInt(epoch, 10)
+}
 
-		// the date below is how the golang date formatter works. it's used for the formatting. it's not what is actually going to be displayed
-		return time.Now().Add(time.Duration(offsetDays) * 24 * time.Hour).Format("2006-01-02")
+func randomElements(source string) string {
+	r := templateElementsRegex.FindStringSubmatch(source)
+
+	if r == nil {
+		return source
+	}
+
+	s := strings.Split(r[1], ",")
+	number := rand.Intn(len(s))
+
+	return s[number]
+}
+
+func rangeElements(source string) string {
+	r := templateRangeRegex.FindStringSubmatch(source)
+	if r == nil {
+		return source
+	}
+
+	min, _ := strconv.Atoi(r[1])
+	max, _ := strconv.Atoi(r[2])
+
+	number := rand.Intn(max-min+1) + min
```

Comment:
Do we need random/different values for each run (i.e. do we need to use the [Seed function](https://golang.cafe/blog/golang-random-number-generator.html)).